Code/Legacy/CrySystem/Log.cpp

```diff
@@ -897,128 +980,118 @@ void CLog::LogStringToFile(const char* szString, ELogType logType, bool bAdd, [[
-            strftime(sTime, 20, "<%H:%M:%S> ", &today);
+            strftime(sTime, TimeStringType::max_size(), "<%Y-%m-%d %H:%M:%S> ", &today);
 #else
             auto today = localtime(&ltime);
-            strftime(sTime, 20, "<%H:%M:%S> ", today);
+            strftime(sTime, TimeStringType::max_size(), "<%Y-%m-%d %H:%M:%S> ", today);
 #endif
-            tempString = LogStringType(sTime) + tempString;
+            return sTime;
+        };
 
+        auto GetElapsedTimeInSeconds = []() -> TimeStringType
+        {
+            TimeStringType elapsedTime;
             static AZ::TimeMs lasttime = AZ::Time::ZeroTimeMs;
             const AZ::TimeMs currenttime = AZ::GetRealElapsedTimeMs();
             if (lasttime != AZ::Time::ZeroTimeMs)
             {
-                timeStr.clear();
-                uint32 dwMs = (uint32)(currenttime - lasttime);
-                timeStr = AZStd::string::format("<%3d.%.3d>: ", dwMs / 1000, dwMs % 1000);
-                tempString = timeStr + tempString;
+                uint32 dwMs = aznumeric_cast<uint32>(currenttime - lasttime);
+                elapsedTime = TimeStringType::format("<%3d.%.3d>: ", dwMs / 1000, dwMs % 1000);
             }
             lasttime = currenttime;
-        }
-        else if (dwCVarState == 4)             // Log_IncludeTime
+
+            return elapsedTime;
+        };
+
+        auto GetElapsedTimeSinceStartInSeconds = []() -> TimeStringType
         {
-            static bool bFirst = true;
+            TimeStringType elapsedTime;
 
             static AZ::TimeMs lasttime = AZ::Time::ZeroTimeMs;
             const AZ::TimeMs currenttime = AZ::GetRealElapsedTimeMs();
             if (lasttime != AZ::Time::ZeroTimeMs)
             {
-                timeStr.clear();
                 uint32 dwMs = (uint32)(currenttime - lasttime);
-                timeStr = AZStd::string::format("<%3d.%.3d>: ", dwMs / 1000, dwMs % 1000);
-                tempString = timeStr + tempString;
+                elapsedTime = TimeStringType::format("<%3d.%.3d>: ", dwMs / 1000, dwMs % 1000);
             }
+
+            static bool bFirst = true;
             if (bFirst)
             {
                 lasttime = currenttime;
                 bFirst = false;
             }
+
+            return elapsedTime;
+        };
+
+        if (dwCVarState == 1)              // Log_IncludeTime
+        {
+            timeStr = GetHourMinuteSeconds();
+        }
+        else if (dwCVarState == 2)     // Log_IncludeTime
+        {
+            timeStr = GetElapsedTimeInSeconds();
+        }
+        else if (dwCVarState == 3)     // Log_IncludeTime
+        {
+            timeStr = GetHourMinuteSecond();
+            timeStr += GetElapsedTimeInSeconds();
+        }
+        else if (dwCVarState == 4)             // Log_IncludeTime
+        {
+            timeStr = GetElapsedTimeSinceStartInSeconds();
         }
         else if (dwCVarState == 5)             // Log_IncludeTime
```

Comment:
uber nit: And chance we can left align the comments or at least have them all line up 😝 

e.g.

```
else if (dwCVarState == 4) // Log_IncludeTime
```

Reply:
How about removing the comments?
I don't see their usefulness in the first place.

Reply:
That works too! 😅 Good idea 🙂